mne/io/base.py

```diff
@@ -212,14 +212,56 @@ def _check_fun(fun, d, *args, **kwargs):
 
 class _BaseRaw(ProjMixin, ContainsMixin, PickDropChannelsMixin,
                SetChannelsMixin, InterpolationMixin, ToDataFrameMixin):
-    """Base class for Raw data"""
+    """Base class for Raw data
+
+    Subclasses must provide the following methods:
+
+        * _read_segment(start, stop, sel, projector, verbose)
+          (only needed for types that support on-demand disk reads)
+    """
     @verbose
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError
+    def __init__(self, info, data=None, cals=None,
+                 first_samps=(0,), last_samps=None,
+                 filenames=(), rawdirs=(),
+                 comp=None, orig_comp_grade=None,
+                 orig_format='double', verbose=None):
+        self.info = info
+        self._data = data
+        if cals is None:
+            cals = np.empty(info['nchan'])
+            for k in range(info['nchan']):
+                cals[k] = info['chs'][k]['range'] * info['chs'][k]['cal']
```

Comment:
I think this is usually not much of an issue because for almost all formats `range == 1`. I'm fine with unifying it so we never store `_cals` at all, just make sure that it won't slow down the raw reading and writing. I suppose it shouldn't, since recalculating it is just a list comprehension.